src/modules/content-s3/content-s3.c

```diff
@@ -50,7 +48,61 @@ static void content_s3_destroy (struct content_s3 *ctx)
     s3_cleanup();
 }
 
-static int parse_config (struct content_s3 *ctx,
+static int parse_credentials (struct s3_config *cfg,
+                              const char *cred_file,
+                              char *errbuff,
+                              int eb_size)
+{
+    struct tomltk_error toml_error;
+    toml_table_t *tbl;
+    const char *raw;
+    char *access_key;
+    char *secret_key;
+
+    if (!(tbl = tomltk_parse_file (cred_file, &toml_error))) {
+        snprintf(errbuff, eb_size, "toml parse failed in %d: %s", 
+                 toml_error.lineno, toml_error.errbuf);
+        goto error;
+    }
+
+    if (!(raw = toml_raw_in(tbl, "secret-access-key"))) {
+        snprintf(errbuff, eb_size, "failed to parse secret key");
+        goto error;
+    }
+
+    if (toml_rtos (raw, &secret_key)) {
+        snprintf(errbuff, eb_size, "failed to parse secret key");
+        goto error;
+    }
+
+    if (!(raw = toml_raw_in(tbl, "access-key-id"))) {
+        snprintf(errbuff, eb_size, "failed to parse access key");
+        goto error;
+    }
+
+    if (toml_rtos (raw, &access_key)) {
+        snprintf(errbuff, eb_size, "failed to parse access key");
+        goto error;
+    }
+
+    cfg->secret_key = secret_key;
+    cfg->access_key = access_key;
+
```

Comment:
You're missing a `toml_free (tbl)` .
Also if access key fetch fails, need to free the secret key.